codec/proto_codec_test.go

```diff
@@ -43,15 +41,6 @@ func TestProtoCodec(t *testing.T) {
 	testMarshaling(t, cdc)
 }
 
-type lyingProtoMarshaler struct {
-	codec.ProtoMarshaler
-	falseSize int
-}
-
-func (lpm *lyingProtoMarshaler) Size() int {
-	return lpm.falseSize
-}
-
```

Comment:
since we are proto.Size() now, this test isn't applicable anymore. The downside is we are taking an extra performance hit on the proto.Size call. I think we could optimise our own, to return the marshalled value + the size in order to avoid marshalling twice here.

Reply:
removed size functions to avoid extra marshalling